src/Sylius/Behat/Context/Setup/PaymentContext.php

```diff
@@ -49,15 +49,17 @@ public function __construct(RepositoryInterface $paymentMethodRepository, Shared
     }
 
     /**
-     * @Given store allows paying offline
+     * @Given store allows paying :paymentMethodName
      */
-    public function storeAllowsPayingOffline()
+    public function storeAllowsPaying($paymentMethodName)
     {
-        $paymentMethod = $this->paymentMethodFactory->createNew();
-        $paymentMethod->setCode('PM1');
-        $paymentMethod->setGateway('offline');
-        $paymentMethod->setName('Offline');
-        $paymentMethod->setDescription('Offline payment method');
+        $properties = [
+            'code' => 'PM1',
+            'name' => $paymentMethodName,
+            'description' => 'Payment method',
+        ];
+
+        $paymentMethod = $this->paymentMethodFactory->createFromArray($properties);
```

Comment:
Not sure about this change, why? It would make more sense if we process a TableNode here, but we are not.